### PR TITLE
🚨(project) resolve dockerfile build warnings

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

Since a recent version of Docker, many warnings are raised at build time from the Dockerfiles.
Fixing them.

### CODE REVIEW

File: src/api/Dockerfile [hunk 1]
```diff
@@ -65,15 +65,17 @@ RUN pip install -e ./core[dev] && \
     done
 
 # Un-privileged user running the application
-USER ${DOCKER_USER:-1000}
+ARG DOCKER_USER
```
Comment (claudusd) [inline src/api/Dockerfile:68]: You should defined a default value for the [`ARG`](https://docs.docker.com/reference/dockerfile/#arg)
```suggestion
ARG DOCKER_USER=1000
```

File: src/app/Dockerfile [hunk 2]
```diff
@@ -50,10 +50,11 @@ RUN mkdir -p /usr/local/etc/gunicorn
 COPY ./docker/files/usr/local/etc/gunicorn/warren.py /usr/local/etc/gunicorn/warren.py
 
 # Un-privileged user running the application
-USER ${DOCKER_USER:-1000}
+ARG DOCKER_USER
```
Comment (claudusd) [inline src/app/Dockerfile:53]:
```suggestion
ARG DOCKER_USER=1000
```

File: src/app/Dockerfile [hunk 3]
```diff
@@ -64,13 +65,14 @@ RUN pip uninstall -y warren
 RUN pip install -e .[dev]
 
 # Switch back to Un-privileged user to run the application
-USER ${DOCKER_USER:-1000}
+ARG DOCKER_USER
```
Comment (claudusd) [inline src/app/Dockerfile:68]:
```suggestion
ARG DOCKER_USER=1000
```

File: src/frontend/Dockerfile [hunk 4]
```diff
@@ -32,19 +32,21 @@ WORKDIR /app
...
+FROM core AS development
 
 # Install Git (required for changeset)
 RUN apt-get update && \
     apt-get install -y git && \
     rm -rf /var/lib/apt/lists/*
 
 # Un-privileged user running the application
-USER ${DOCKER_USER:-1000}
+ARG DOCKER_USER
```
Comment (claudusd) [inline src/frontend/Dockerfile:43]:
```suggestion
ARG DOCKER_USER=1000
```

File: src/frontend/Dockerfile [hunk 5]
```diff
@@ -32,19 +32,21 @@ WORKDIR /app
...
+USER ${DOCKER_USER}
 
 
 # -- Production --
-FROM core as production
+FROM core AS production
 
 # Un-privileged user running the application
-USER ${DOCKER_USER:-1000}
+ARG DOCKER_USER
```
Comment (claudusd) [inline src/frontend/Dockerfile:51]:
```suggestion
ARG DOCKER_USER=1000
```

**Review by claudusd — CHANGES_REQUESTED** [review]

File: src/api/Dockerfile (hunk 1 above)
Comment (wilbrdt) [reply src/api/Dockerfile]: you're right, thanks!

**Review by claudusd — APPROVED** [review]

**Pull request merged**